### PR TITLE
tech detection: Maintenance tweaks

### PR DESCRIPTION
**Pull request by kingthorin**
## Overview
- Apply static modifier where applicable, and access things in a static manner.
- Drop 'wappalyzer' from class names where practical.

## Related Issues
N/A

## Checklist
- [na] Update help
- [x] Update changelog
- [x] Run `./gradlew spotlessApply` for code formatting
- [na] Write tests
- [na] Check code coverage
- [x] Sign-off commits
- [x] Squash commits
- [x] Use a descriptive title

### CODE REVIEW

File: addOns/wappalyzer/CHANGELOG.md [hunk 1]
```diff
@@ -10,6 +10,7 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/).
     - Quick > Return on first match; which may mean missing version information, but should be slightly more performant. (This is the default.)
     - Exhaustive > Keep matching and don't return early; likely slightly less performant.
 - The add-on now also has the ability to optionally raise Alerts for each technology identified. The default setting is enabled. (Issue 8361)
+- Maintenance changes.
```
Comment (thc202) [inline addOns/wappalyzer/CHANGELOG.md:13]: Better mention that these changes might break existing code using the AF tech data.

File: addOns/wappalyzer/src/main/java/org/zaproxy/zap/extension/wappalyzer/TechAPI.java [hunk 2]
```diff
@@ -31,7 +31,7 @@
 import org.zaproxy.zap.extension.api.ApiResponseSet;
 import org.zaproxy.zap.extension.api.ApiView;
 
-public class WappalyzerAPI extends ApiImplementor {
+public class TechAPI extends ApiImplementor {
```
Comment (thc202) [inline addOns/wappalyzer/src/main/java/org/zaproxy/zap/extension/wappalyzer/TechAPI.java:34]: `TechApi`

Comment (thc202) [reply addOns/wappalyzer/src/main/java/org/zaproxy/zap/extension/wappalyzer/TechAPI.java:34]: This should also be updated in `wappalyzer.gradle.kts`.

**Comment by kingthorin** [comment]
Done, done, & done 👍

**Review by thc202 — APPROVED** [review]

**Comment by thc202** [comment]
Thank you!

**Review by psiinon — APPROVED** [review]

**Pull request merged**